envoyextensions/extensioncommon/basic_envoy_extender.go

```diff
@@ -111,90 +111,21 @@ func (b *BasicEnvoyExtender) Extend(resources *xdscommon.IndexedResources, confi
 	return resources, resultErr
 }
 
-func (b *BasicEnvoyExtender) patchListener(config *RuntimeConfig, l *envoy_listener_v3.Listener) (proto.Message, bool, error) {
+func (b *BasicEnvoyExtender) patchSupportedListenerFilterChains(config *RuntimeConfig, l *envoy_listener_v3.Listener) (proto.Message, bool, error) {
 	switch config.Kind {
-	case api.ServiceKindTerminatingGateway:
-		return b.patchTerminatingGatewayListener(config, l)
-	case api.ServiceKindConnectProxy:
-		return b.patchConnectProxyListener(config, l)
+	case api.ServiceKindTerminatingGateway, api.ServiceKindConnectProxy:
+		return b.patchListenerFilterChains(config, l)
 	}
 	return l, false, nil
 }
```

Comment:
After double checking the original [change](https://github.com/hashicorp/consul/commit/21c31345752baf68b8b8071eeae0957cef8a0937) that introduced this switch and the matching one above, I restored it to align with that and reduce changes here. This could prevent a future bug if we were to support more proxy types for extensions in general without aligning that support with listeners specifically.

No other changes since the last review.